instruqt/protecting-your-assets/06-validating-license-fields/assignment.md

```diff
@@ -0,0 +1,188 @@
+---
+slug: validating-license-fields
+id: xhamc4jj1gl1
+type: challenge
+title: Validating License Fields with the Replicated SDK
+teaser: Calling Replicated SDK to assure customer entitlements
+notes:
+- type: text
+  contents: Learn how to validate the entitlements we just created
+tabs:
+- title: Shell
+  type: terminal
+  hostname: shell
+  workdir: /home/replicant
+difficulty: basic
+timelimit: 600
+---
+
+The Replicated SDK provides an in-cluster API for the Replicated platform. One
+of it's core features is access to the complete customer license. We're going
+to run some simple shell commands against the SDK to show how your team can
+validating entitlements as part of your application.
+
+Everything we're doing in this lab you would generally do as part of your
+application code. We're using shell commands here since Replicated customers
+work with many different programming languages, but everyone knows `curl` and
+it's friends.
+
+The Replicated SDK
+==================
+
+The SDK runs alongside your application and enables access to the Replicated
+Platform without having to make calls outside of the cluster. It in turn
+connects to the vendor portal periodically to check for update. Each update
+check includes telemetry information we use to help you better understand you
+customer's installation. See [Closing the Support Information Gap](https://play.instruqt.com/replicated/tracks/closing-infromation-gap) for more information.
+
+The SDK is your mechanism for interacting with the platform from your
+application. You can use it to get information about the currently installed
+version, check for updates, and interact with the license. You can also send
+custom metrics that supplement the default telemetry. Slackernews already
+includes the Replicated SDK, so we won't need to add it to the chart. To learn
+more about including it in your application, try the [Distributing Your
+Application with
+Replicated](https://play.instruqt.com/replicated/tracks/distributing-with-replicated)
+lab.
+
+Interacting with the SDK
+========================
+
+Since we're going to shell commands to call the SDK, let's use the "Netshoot"
+image from Nicola Kabar. This image contains a lot of network troubleshooting
+tools and tools you need to work with them. That means it has `curl`, `jq`, and
+other commands that make it easier to work with APIs from the command line.
+
+```
+kubectl run -n slackernews sdk-client -it --image nicolaka/netshoot
+```
+
+Let's get the license fields for the Slackernews application.
+
+```
+curl http://replicated:3000/api/v1/license/fields | jq .
```

Comment:
```suggestion
curl -s http://replicated:3000/api/v1/license/fields | jq .
```

The curl output itself is confusing come the last step. I ran it like 3 times before I noticed the `100` that was returned was the value, b/c it printed right next to a `100` that was download compleation. We can avoid that by just silenceing curl so people can focus on the result.